python_modules/dagster/dagster/core/host_representation/repository_location.py

```diff
@@ -289,7 +289,9 @@ def executable_path(self) -> Optional[str]:
 
     @property
     def container_image(self) -> str:
-        return self._recon_repo.container_image
+        # TODO: `container_image` can be None, not clear if the type annotation
+        # or method logic here is wrong
```

Comment:
Did you try updating the type hint to Optional[str] to see what broke? I would feel better if this type hint was accurate or if there is some assumption that this is never `None` in this case that we throw in this getter 

maybe address this in a follow up since there is so much in this diff

Reply:
I tracked down the access sites for `container_image` and there are checks for `None`, so I'm pretty sure the annotation was wrong, I've updated accordingly.